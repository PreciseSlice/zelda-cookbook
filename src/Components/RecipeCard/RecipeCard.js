// eslint-disable-next-line
import React from 'react';
import './RecipeCard.css';
import PropTypes from 'prop-types';
import heart from '../../assets/heart.png';
import greenRupee from '../../assets/GreenRupee.png';

const RecipeCard = (props) => {
  const {
    id,
    image,
    name, 
    hearts, 
    type, 
    duration, 
    notes, 
    ingredient1,
    ingredient1Image, 
    ingredient2,
    ingredient2Image,    
    ingredient3,
    ingredient3Image,     
    ingredient4, 
    ingredient4Image,     
    ingredient5, 
    ingredient5Image,     
    strength, 
    resale
  } = props;

  const clicked = (e) => {
    const recipeTarget = e.target;
    const targetArticle = recipeTarget.closest('article');
    if(targetArticle.className.includes('active-recipe')) {
      targetArticle.classList.remove('active-recipe');
    } else {
      targetArticle.classList.add('active-recipe');  
    }
  };
  
  return (
    <div className="recipe-card-container">
      <article id={ id } className={ name } onClick={clicked} >

        <div className={`${ name } recipe-details`}>
          <img src={ image } alt={ name } />
          <div className={`${ name } name-hearts`}>
            <h2 className={`${ name } recipe-name`}>{ name }</h2>
            <h3 className={`${ name } recipe-hearts`}><img src={ heart } alt="Number Of Hearts"/><span id="targetHeart">{ hearts }</span></h3>
          </div>
          <h5 className={`${ name } recipe-type`}>Type: { type }</h5>
        </div>

        <h3 className={`${ name } recipe-notes`}>{ notes === null ? '' : 'Notes: ' +  notes }</h3>

        <div className={`${ name } recipe-specific-details`}>
          <h5 className={`${ name } strength`}>Strength: { strength }</h5>
          <h5 className={`${ name } duration`}>Duration: { duration }</h5>
          <h5 className={`${ name } resale`}><img src={ greenRupee } alt="Green Rupees Jewel"/><p>{ resale }</p></h5>
        </div>

        <div className={`${ name } ingredients`}>
          <h3 className={`${ name } ingredient1`}><img src={ ingredient1Image } alt={ name } />{ ingredient1 }</h3>
          <h3 className={`${ name } ingredient2`}><img src={ ingredient2Image } alt={ name } />{ ingredient2 }</h3>
          <h3 className={`${ name } ingredient3`}><img src={ ingredient3Image } alt={ name } />{ ingredient3 }</h3>
          <h3 className={`${ name } ingredient4`}><img src={ ingredient4Image } alt={ name } />{ ingredient4 }</h3>
          <h3 className={`${ name } ingredient5`}><img src={ ingredient5Image } alt={ name } />{ ingredient5 }</h3>
        </div>
      </article>
    </div>
  );
};

const { number, string, func } = PropTypes;

RecipeCard.propTypes = {
  id: number,
  name: string, 
  hearts: string, 
  type: string, 
  duration: string, 
  notes: string, 
  ingredient1: string, 
  ingredient2: string, 
  ingredient3: string, 
  ingredient4: string, 
  ingredient5: string,
  ingredient1_id: number, 
  ingredient2_id: number, 
  ingredient3_id: number, 
  ingredient4_id: number, 
  ingredient5_id: number, 
  strength: string, 
  resale: string,
  clicked: func
};

export default RecipeCard;
