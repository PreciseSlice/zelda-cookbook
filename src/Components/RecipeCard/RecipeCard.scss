@import url('https://fonts.googleapis.com/css?family=PT+Serif:700');

#all-recipe-container {
  display: flex;
  flex-direction: column;
  font-family: 'PT Serif', serif;
  overflow-y: scroll;
  height: 100vh;
}

.recipe-card-container {
  min-width: 558px;  
  max-width: 558px;
  margin: 10px auto;
  border-radius: 10px;
  background-color: rgba(88,98,94,.75);
  padding: 10px;
  color: rgb(223, 213, 195);
  position: relative;
}

article {
  height: 88px;
  overflow-y: hidden
}

.active-recipe {
  height: fit-content;
}

.recipe-details {
  display: flex;
  flex-direction: row;
}

.name-hearts {
  display: flex;
  flex-direction: column;
}

.recipe-name,
.recipe-hearts {
  margin-top: 0px;
  margin-bottom: 5px;
  padding-left: 20px;
}

.recipe-hearts {
  display: grid; 
  grid-template-columns: 35px 1fr;
}

.recipe-hearts img {
  align-self: center;
  margin-right: 10px;
  width: 25px;
  height: 20px;
}

#targetHeart {
  align-self: start;
}

.recipe-type {
  margin-top: 5px;
  padding-left: 10px;
  margin-left: auto;
}

img {
  width: 80px;
  height: 80px;
}

.resale {
  display: flex;
  flex-direction: row;
}

.resale img {
  width: 15px;
  height: 24px;
}

.resale p {
  margin: 3px;
  padding-left: 7px;
}

.ingredients,
.recipe-specific-details {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
}

.type_image {
  height: 30px;
  width: 30px;
  background-color: black;
  position: absolute;
  right: 10px;
  top: 10px
}