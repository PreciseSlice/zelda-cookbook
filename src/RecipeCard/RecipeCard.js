import React from 'react';
import './RecipeCard.css';
// import PropTypes from 'prop-types';

const RecipeCard = (props) => {
    const {id,
          name, 
          hearts, 
          type, 
          duration, 
          notes, 
          ingredient1, 
          ingredient2, 
          ingredient3, 
          ingredient4, 
          ingredient5, 
          strength, 
          resale } = props
  
  const recipeHtml = <div className="recipe-card" id={ id }>
                        <img src="../assets/zelda-header.png" alt="Seafood Curry" />
                        <h1 className="recipe-name">Name: { name }</h1>
                        <h3 className="recipe-hearts">Hearts: { hearts }</h3>
                        <div className="recipe-details">
                          <h5 className="recipe-type">Type: { type }</h5>
                          <h5 className="strength">Strength: { strength }</h5>
                          <h5 className="duration">Duration: { duration }</h5>
                          <h5 className="resale">Resale: { resale }</h5>
                        </div>
                        <h3 className="recipe-notes">Notes: { notes }</h3>
                        <div className="ingredients">
                          <h3 className="ingredient1">{ ingredient1 ? ingredient1 : 'N/A'}</h3>
                          <h3 className="ingredient2">{ ingredient2 ? ingredient2 : 'N/A'}</h3>
                          <h3 className="ingredient3">{ ingredient3 ? ingredient3 : 'N/A'}</h3>
                          <h3 className="ingredient4">{ ingredient4 ? ingredient4 : 'N/A'}</h3>
                          <h3 className="ingredient5">{ ingredient5 ? ingredient5 : 'N/A'}</h3>
                        </div>
                     </div>        

  return (
    <div id="all-recipes">
      { recipeHtml }
    </div>
  )
}






export default RecipeCard;
