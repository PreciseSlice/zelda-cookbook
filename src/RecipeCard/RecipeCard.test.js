import React from 'react';
import { shallow } from 'enzyme';
import RecipeCard from './RecipeCard';

describe('RecipeCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RecipeCard
      id = {106}
      name = 'Baked Apple'
      hearts = 'Â¾'
      type = 'Restore Hearts'
      duration = {null}
      notes = 'Open flame.'
      ingredient1 = {2}
      ingredient2 = {null}
      ingredient3 = {null}
      ingredient4 = {null}
      ingredient5 = {null}
      strength = {null}
      resale = '0'
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
});