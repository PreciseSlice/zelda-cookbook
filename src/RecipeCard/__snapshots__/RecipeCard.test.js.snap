// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard should match the snapshot 1`] = `
<div
  id="recipe-card-container"
>
  <article
    className="Baked Apple"
    onClick={[Function]}
  >
    <div
      className="Baked Apple recipe-details"
    >
      <img
        alt="Seafood Curry"
        src="Seafoodcurry.jpg"
      />
      <div
        className="Baked Apple name-hearts"
      >
        <h2
          className="Baked Apple recipe-name"
        >
          Name: 
          Baked Apple
        </h2>
        <h3
          className="Baked Apple recipe-hearts"
        >
          Hearts: 
          Â¾
        </h3>
      </div>
      <h5
        className="Baked Apple recipe-type"
      >
        Type: 
        Restore Hearts
      </h5>
    </div>
    <h3
      className="Baked Apple recipe-notes"
    >
      Notes: 
      Open flame.
    </h3>
    <div
      className="Baked Apple recipe-specific-details"
    >
      <h5
        className="Baked Apple strength"
      >
        Strength: 
      </h5>
      <h5
        className="Baked Apple duration"
      >
        Duration: 
      </h5>
      <h5
        className="Baked Apple resale"
      >
        <img
          alt="Green Rupees Jewel"
          src="GreenRupee.png"
        />
        <p>
          0
        </p>
      </h5>
    </div>
    <div
      className="Baked Apple ingredients"
    >
      <h3
        className="Baked Apple ingredient1"
      >
        2
      </h3>
      <h3
        className="Baked Apple ingredient2"
      />
      <h3
        className="Baked Apple ingredient3"
      />
      <h3
        className="Baked Apple ingredient4"
      />
      <h3
        className="Baked Apple ingredient5"
      />
    </div>
  </article>
</div>
`;
